### DISCUSSION TITLE
mocking

### DISCUSSION THREAD

**Discussion by Silver343** [stefanzweifel/sidecar-browsershot#33] [Q&A]
Has anyone mocked the browshershot lambda class, I am attempted a partial mock but it returns `  Static method Mockery_2_Wnx_SidecarBrowsershot_BrowsershotLambda::html() does not exist on this mock object`

here is my code to make the mock
 `
$this->partialMock(BrowsershotLambda::class)->shouldReceive('saveToS3')->andReturn(True);`

**Comment by stefanzweifel** [comment]
In my projects, I usually add a Container Binding to my `AppServiceProvider` to only use `BrowsershotLambda` in production.
The code looks like this.

```php
use Spatie\Browsershot\Browsershot;
use Wnx\SidecarBrowsershot\BrowsershotLambda;

if ($this->app->environment(['production'])) {
    $this->app->bind(Browsershot::class, BrowsershotLambda::class);
}
```

My tests then actually run Browsershot.

---

I don't use Mocks that often, but I think the issue could be that `html()` is not on `BrowsershotLambda` but on the `Browsershot` class from Spatie. `BrowsershotLambda` extends `Browsershot`.
https://github.com/stefanzweifel/sidecar-browsershot/blob/main/src/BrowsershotLambda.php#L13
